skimage/restoration/_cycle_spin.py

```diff
@@ -143,8 +148,11 @@ def _run_one_shift(shift):
         tmp = func(xs, **func_kw)
         return _roll_axes(tmp, -np.asarray(shift))
 
+    if not dask_available:
+        warn('The optional dask dependency is not installed. '
+             'The number of worker is set to 1.')
```

Comment:
imho this warning should only be triggered *if* dask is not available *and* the number of workers is set to a value other than 1. Which, incidentally, you're not setting here! So I think you need to

- change the condition to also check whether `num_workers` is set to `>1` or `None`
- set `num_workers = 1` inside the condition
- add a test for all three conditions, including a test for `num_workers=1` *not* raising a warning.

Reply:
oh, that's what I wanted to do! I forgot half of the condition... :( I'll fix it.

Reply:
@sciunto 

a) "number of worker*s* is set to 1"
b) could you add a test to verify that `num_workers=1` results in no warning being raised
c) could you expand your message to say, "to silence this warning, install dask or explicitly set `num_workers=1` when calling the `cycle_spin` function"?

Thanks!

Reply:
I think all your requests are satisfied. I don't think we need an extra test.